modules/apps/portal-search/portal-search-admin-web/src/main/java/com/liferay/portal/search/admin/web/internal/portlet/action/EditMVCActionCommand.java

```diff
@@ -211,41 +209,30 @@ private void _reindexIndexReindexer(ActionRequest actionRequest)
 					ParamUtil.getString(actionRequest, "executionMode")));
 		}
 
-		IndexReindexer indexReindexer =
-			_indexReindexerRegistry.getIndexReindexer(className);
+		ThemeDisplay themeDisplay = (ThemeDisplay)actionRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
 
-		indexReindexer.reindex(
-			ParamUtil.getLongValues(actionRequest, "companyIds"),
-			ParamUtil.getString(actionRequest, "executionMode"));
+		_indexWriterHelper.reindex(
```

Comment:
i was originally thinking we would just call _backgroundTaskManager.addBackgroundTask right here. can you try that instead of using _indexWriterHelper?

also, you need to pass the execution mode in the taskContextMap